Develop stream 2025-03-10 (#622)
ROCm/rocRAND: CHANGELOG.md

  * `rocrand_h_sobol64_direction_vectors`, use `rocrand_get_direction_vectors64` instead.
  * `rocrand_h_scrambled_sobol32_direction_vectors`, use `rocrand_get_direction_vectors32` instead.
  * `rocrand_h_scrambled_sobol64_direction_vectors`, use `rocrand_get_direction_vectors64` instead.

> * Deprecated rocRAND's Fortran API in favor of hipfort.

Reviewer: Items that are removed need to be moved to the ### Removed section.
If the deprecated item is being removed in 6.5, then it should also be in removed. If it's a notification that it will be removed in a later release, then it should go into ###Upcoming changes

> * Deprecated rocRAND's Fortran API in favor of hipfort.

Reviewer: ```suggestion
* Deprecated the rocRAND Fortran API in favor of hipfort.
```

> * Removed inline assembly and the `ENABLE_INLINE_ASM` CMake option. Inline assembly was used for optimization of multiplications in the Mrg32k3a and Philox 4x32-10 generators, it is no longer needed because the current HIP compiler is able to produce code with the same or better performance (rare regressions are smaller than 2%).

Reviewer: ```suggestion
* Removed inline assembly and the `ENABLE_INLINE_ASM` CMake option. Inline assembly was used to optimize multiplications in the Mrg32k3a and Philox 4x32-10 generators. It is no longer needed because the current HIP compiler is able to produce code with the same or better performance (rare regressions are smaller than 2%).
```

> * Removed inline assembly and the `ENABLE_INLINE_ASM` CMake option. Inline assembly was used for optimization of multiplications in the Mrg32k3a and Philox 4x32-10 generators, it is no longer needed because the current HIP compiler is able to produce code with the same or better performance (rare regressions are smaller than 2%).

Reviewer: Do we need to include the aside about the rare regressions? I think this might be too much detail for a changelog entry.

> * Removed usages of the deprecated clang definition `__AMDGCN_WAVEFRONT_SIZE`.

Reviewer: ```suggestion
* Removed instances of the deprecated clang definition `__AMDGCN_WAVEFRONT_SIZE`.
```
